File: ICPC_Mexico_SegundaFecha/K.py
n = int(input())
c = [int(i) for i in input().split()]
k = [int(i) for i in input().split()]
c.sort()
k.sort()

count = 0
flag = False

for i in range(n):
    if (k[i] > c[i]):
        print(-1)
        flag = True
    elif i==0:
        count += k[i]
    else: 
        count += c[i]

if not flag: print(count)

File: ICPC_Mexico_SegundaFecha/K.cpp
#include <bits/stdc++.h>
#define endl "\n"

using namespace std;
typedef long long ll;
typedef long double ld;
typedef pair<int, int> paii;
typedef vector<int> vi;

int const MAXN = 2e5+5;
pair<ll, ll> p[MAXN];

int main() {
   ios::sync_with_stdio(0);
   cin.tie(0);
   int n; cin >> n;
   ll asked = 0;
   for (int i = 0; i < n; i++) cin >> p[i].first;
   for (int i = 0; i < n; i++) cin >> p[i].second;

   sort(p, p+n);

   ll count = 0;

   for (int i = n-1; i >= 0; i--){
      if (p[i].second > p[i].first){
         cout << -1 << endl;
         return 0;
      }
      else if (i == 0) count += p[i].second;
      else count += p[i].first;
   }
   cout << count << endl;
   return 0;
}
